joblib/test/test_func_inspect.py

```diff
@@ -81,58 +70,69 @@ def f(self, x):
+              (i, [[], (2, )], {'x': 2})])
+def test_filter_args(func, args, filtered_args):
+    assert filter_args(func, *args) == filtered_args
 
 
 def test_filter_args_method():
     obj = Klass()
     assert filter_args(obj.f, [], (1, )) == {'x': 1, 'self': obj}
 
 
-def test_filter_varargs():
-    yield assert_equal, filter_args(h, [], (1, )), \
-                            {'x': 1, 'y': 0, '*': [], '**': {}}
-    yield assert_equal, filter_args(h, [], (1, 2, 3, 4)), \
-                            {'x': 1, 'y': 2, '*': [3, 4], '**': {}}
-    yield assert_equal, filter_args(h, [], (1, 25), dict(ee=2)), \
-                            {'x': 1, 'y': 25, '*': [], '**': {'ee': 2}}
-    yield assert_equal, filter_args(h, ['*'], (1, 2, 25), dict(ee=2)), \
-                            {'x': 1, 'y': 2, '**': {'ee': 2}}
+@parametrize(['func', 'args', 'filtered_args'],
+             [(h, [[], (1, )],
+               {'x': 1, 'y': 0, '*': [], '**': {}}),
+              (h, [[], (1, 2, 3, 4)],
+               {'x': 1, 'y': 2, '*': [3, 4], '**': {}}),
+              (h, [[], (1, 25), dict(ee=2)],
+               {'x': 1, 'y': 25, '*': [], '**': {'ee': 2}}),
+              (h, [['*'], (1, 2, 25), dict(ee=2)],
+               {'x': 1, 'y': 2, '**': {'ee': 2}})])
+def test_filter_varargs(func, args, filtered_args):
+    assert filter_args(func, *args) == filtered_args
 
 
-def test_filter_kwargs():
-    assert (filter_args(k, [], (1, 2), dict(ee=2)) ==
-            {'*': [1, 2], '**': {'ee': 2}})
-    assert filter_args(k, [], (3, 4)) == {'*': [3, 4], '**': {}}
+@parametrize(['func', 'args', 'filtered_args'],
+             [(k, [[], (1, 2), dict(ee=2)],
+               {'*': [1, 2], '**': {'ee': 2}}),
+              (k, [[], (3, 4)],
+               {'*': [3, 4], '**': {}})])
+def test_filter_kwargs(func, args, filtered_args):
+    assert filter_args(func, *args) == filtered_args
 
 
 def test_filter_args_2():
     assert (filter_args(j, [], (1, 2), dict(ee=2)) ==
             {'x': 1, 'y': 2, '**': {'ee': 2}})
 
-    assert_raises(ValueError, filter_args, f, 'a', (None, ))
-    # Check that we capture an undefined argument
-    assert_raises(ValueError, filter_args, f, ['a'], (None, ))
     ff = functools.partial(f, 1)
     # filter_args has to special-case partial
     assert filter_args(ff, [], (1, )) == {'*': [1], '**': {}}
     assert filter_args(ff, ['y'], (1, )) == {'*': [1], '**': {}}
 
 
-def test_func_name():
-    yield assert_equal, 'f', get_func_name(f)[1]
-    # Check that we are not confused by the decoration
-    yield assert_equal, 'g', get_func_name(g)[1]
+@parametrize(['func', 'funcname'], [(f, 'f'), (g, 'g'), (f2, 'f2')])
+def test_func_name(func, funcname):
+    # Check that we are not confused by decoration
+    # here testcase 'f2' is the function itself
+    assert get_func_name(func)[1] == funcname
+
+
+def test_func_name_on_inner_func(f2):
+    # Check that we are not confused by decoration
```

Comment:
`g` was decorated with `@mem.cache` earlier, and it was the only one function. All other functions - `f`, `f2`, `h`, `i`, `j`, `k` were normal ones. That seemed a bit off to me to I went ahead and made this change.